### ISSUE TITLE
Error: Cannot read property 'object' of undefined when using BrowserWindow.setProgressBar()

### ISSUE THREAD

**Issue by jviotti** [electron/electron#5229]
We got the following anonymous [Etcher](https://github.com/resin-io/etcher) stack trace on TrackJS by a user running GNU/Linux:

```
["Error: Cannot read property 'object' of undefined\nTypeError: Cannot read property 'object' of undefined
at ObjectsRegistry.get (/home/wojciech/Downloads/etcher-release/Etcher-linux-x64/resources/atom.asar/browser/objects-registry.js:44:28)
at EventEmitter.<anonymous> (/home/wojciech/Downloads/etcher-release/Etcher-linux-x64/resources/atom.asar/browser/rpc-server.js:318:31)
at emitMany (events.js:108:13)
at EventEmitter.emit (events.js:182:7)
at EventEmitter.<anonymous> (/home/wojciech/Downloads/etcher-release/Etcher-linux-x64/resources/atom.asar/browser/api/web-contents.js:144:25)
at emitTwo (events.js:87:13)
at EventEmitter.emit (events.js:172:7)
at metaToValue (/home/wojciech/Downloads/etcher-release/Etcher-linux-x64/resources/atom.asar/renderer/api/remote.js:170:13)
at BrowserWindow.remoteMemberFunction [as setProgressBar] (/home/wojciech/Downloads/etcher-release/Etcher-linux-x64/resources/atom.asar/renderer/api/remote.js:110:18)
at set (/home/wojciech/Downloads/etcher-release/Etcher-linux-x64/resources/app.asar/lib/browser/os/window-progress/services/window-progress.js:55:24)
at file:///home/wojciech/Downloads/etcher-release/Etcher-linux-x64/resources/app.asar/lib/browser/app.js:115:29
at /home/wojciech/Downloads/etcher-release/Etcher-linux-x64/resources/app.asar/lib/browser/utils/notifier/services/notifier.js:46:14
at Scope.$emit (/home/wojciech/Downloads/etcher-release/Etcher-linux-x64/resources/app.asar/node_modules/angular/angular.js:17275:33)
at emit (/home/wojciech/Downloads/etcher-release/Etcher-linux-x64/resources/app.asar/lib/browser/utils/notifier/services/notifier.js:64:16)
at /home/wojciech/Downloads/etcher-release/Etcher-linux-x64/resources/app.asar/lib/browser/modules/image-writer.js:163:25
at /home/wojciech/Downloads/etcher-release/Etcher-linux-x64/resources/app.asar/node_modules/angular/angular.js:18953:31
at completeOutstandingRequest (/home/wojciech/Downloads/etcher-release/Etcher-linux-x64/resources/app.asar/node_modules/angular/angular.js:5824:10)", null]
```

The error originates on:

```
/home/wojciech/Downloads/etcher-release/Etcher-linux-x64/resources/atom.asar/browser/objects-registry.js:44:28
```

In this part of the code, particularly:

```js
// https://github.com/electron/electron/blob/master/lib/browser/objects-registry.js#L44

get (id) {
    return this.storage[id].object
}
```

Sadly we don't have a way to reach out to the user, and the only thing we know is that he's running v0.36.11 on a GNU/Linux system. We suspect that it might be due to a non-well tested Linux distribution, or some weirdness from his environment.

We hope Electron maintainers can draw more conclusions about why the object he's referring to is missing from the object registry, and hopefully provide a check to guard against that.

***

* Electron version: v0.36.11
* Operating system: GNU/Linux

**Comment by CharlieHess** [comment]
This is another way to reproduce #4733.

**Comment by zcbenz** [comment]
@jviotti It had been fixed in v0.37.4, but the fix was not back-ported to v0.36.x because it was a relatively large one.

**Comment by jviotti** [comment]
Perfect, thanks a lot!
